iridauploader/progress/upload_status.py

```diff
@@ -14,114 +13,93 @@
 # File name
 STATUS_FILE_NAME = "irida_uploader_status.info"
 
-# Status field for a sequencing run
-STATUS_FIELD = "Upload Status"
-DATE_TIME_FIELD = "Date Time"
-RUN_ID_FIELD = "Run ID"
-IRIDA_INSTANCE_FIELD = "IRIDA Instance"
-MESSAGE_FIELD = "Message"
-
 
 def get_directory_status(directory, required_file_list):
     """
     Gets the directory status based off using 'irida_uploader_status.info' files to track progress
 
+    Returns as soon as it finds the case that applies
+
     :param directory: the directory to search for a run
     :param required_file_list: optional param: a list of required files that
         are required for that run to be considered valid. Example: ['SampleSheet.csv']
     :return: directory and status dictionary
     """
-    result = DirectoryStatus(directory)
-
     # Verify directory is readable
     if not os.access(directory, os.R_OK):
-        result.status = DirectoryStatus.INVALID
-        result.message = 'Directory cannot be read. Please check permissions'
-        return result
+        directory_status = DirectoryStatus(directory)
+        directory_status.status = DirectoryStatus.INVALID
```

Comment:
I'm seeing pretty much every time you instantiate a `DirectoryStatus` a `directory_status.status = DirectoryStatus.WHATEVER` right after.  Do we want to have a status required in the constructor here?  If it's going to be set every time it's instantiated it likely should be a constructor param.

Reply:
Good call, I added both `status` and `message` params to the constructor